DSCResources/MSFT_xIisLogging/MSFT_xIisLogging.psm1

```diff
@@ -0,0 +1,290 @@
+                     
+        [String]
+        [ValidateSet('Hourly','Daily','Weekly','Monthly','MaxSize')]
+        $LogPeriod,
+                
+        [String]
+        [ValidateRange('1048576','4294967295')]
+        $LogTruncateSize,
+        
+        [Boolean]
+        $LoglocalTimeRollover,
+        
+        [String]
+        [ValidateSet('IIS','W3C','NCSA')]
+        $LogFormat
+    )
+    
+        Assert-Module
+
+        $CurrentLogState = Get-TargetResource -LogPath $LogPath
+        
+        # Check LogFormat
+        if ($PSBoundParameters.ContainsKey('LogFormat'))
+        {
+            # Warn if LogFlags are passed in and Current LogFormat is not W3C
+            if ($PSBoundParameters.ContainsKey('LogFlags') -and $LogFormat -ne 'W3C')
+            {
+                Write-Verbose -Message ($LocalizedData.WarningIncorrectLogFormat)
+            }
+            # Warn if LogFlags are passed in and Desired LogFormat is not W3C
+            if($PSBoundParameters.ContainsKey('LogFlags') -and $CurrentLogState.LogFormat -ne 'W3C')
+            {
+                Write-Verbose -Message ($LocalizedData.WarningIncorrectLogFormat)
+            }
+            # Check LogFormat 
+            if ($LogFormat -ne $CurrentLogState.LogFormat)
+            {
+                Write-Verbose -Message ($LocalizedData.VerboseTestTargetFalseLogFormat)
+                return $false 
+            }
+        }
+        
+        # Check LogFlags
+        if ($PSBoundParameters.ContainsKey('LogFlags') -and (-not (Compare-LogFlags -LogFlags $LogFlags)))  
+        {
+            Write-Verbose -Message ($LocalizedData.VerboseTestTargetFalseLogFlags)
+            return $false
+        }
+            
+        # Check LogPath
+        if ($PSBoundParameters.ContainsKey('LogPath') -and ($LogPath -ne $CurrentLogState.LogPath))
+        { 
+            Write-Verbose -Message ($LocalizedData.VerboseTestTargetFalseLogPath)
+            return $false 
+        }
+        
+        # Check LogPeriod
+        if ($PSBoundParameters.ContainsKey('LogPeriod') -and ($LogPeriod -ne $CurrentLogState.LogPeriod))
+        {
+            if ($PSBoundParameters.ContainsKey('LogTruncateSize'))
+            {
+                Write-Verbose -Message ($LocalizedData.WarningLogPeriod)
+            }
+               
+            Write-Verbose -Message ($LocalizedData.VerboseTestTargetFalseLogPeriod)
+            return $false   
+        }
+        
+        # Check LogTruncateSize
+        if ($PSBoundParameters.ContainsKey('LogTruncateSize') -and ($LogTruncateSize -ne $CurrentLogState.LogTruncateSize))
+        {
+            Write-Verbose -Message ($LocalizedData.VerboseTestTargetFalseLogTruncateSize)
+            return $false
+        }
+        
+        # Check LoglocalTimeRollover
+        if ($PSBoundParameters.ContainsKey('LoglocalTimeRollover') -and ($LoglocalTimeRollover -ne ([System.Convert]::ToBoolean($CurrentLogState.LoglocalTimeRollover))))
```

Comment:
This line needs to come in under 100 chars - just separate with a back tick after the -and if I was you.